src/receivers/ping.rs

```diff
@@ -0,0 +1,68 @@
+extern crate regex;
+
+use std::process::Command;
+
+use regex::Regex;
+
+use crate::message::Message;
+
+use super::Collector;
+
+pub struct Pinger {
+    host: String,
+    ping_count: u8,
+}
+
+impl Pinger {
+    pub fn new(host: String, ping_count: u8) -> Pinger {
+        Pinger { host, ping_count }
+    }
+}
+
+impl Collector for Pinger {
+    fn collect(&self) -> Result<Message, String> {
```

Comment:
I tested this and it works on my machine as well. Can you split this function to smaller functions?

Reply:
It would be also good to write tests for this function.